test/PersistentOrderedMap.prop.test.mo

```diff
@@ -0,0 +1,215 @@
+// @testmode wasi
+
+import Map "../src/PersistentOrderedMap";
+import Nat "../src/Nat";
+import Iter "../src/Iter";
+import Debug "../src/Debug";
+import Array "../src/Array";
+
+import Deque "../src/Deque";
+import Buffer "../src/Buffer";
+
+import Suite "mo:matchers/Suite";
+import T "mo:matchers/Testable";
+import M "mo:matchers/Matchers";
+
+import Random2 "mo:base/Random";
+
+let { run; test; suite } = Suite;
+
+class MapMatcher(expected : Map.Map<Nat, Text>) : M.Matcher<Map.Map<Nat, Text>> {
+  public func describeMismatch(actual : Map.Map<Nat, Text>, _description : M.Description) {
+    Debug.print(debug_show (Iter.toArray(Map.entries(actual))) # " should be " # debug_show (Iter.toArray(Map.entries(expected))))
+  };
+
+  public func matches(actual : Map.Map<Nat, Text>) : Bool {
+    Iter.toArray(Map.entries(actual)) == Iter.toArray(Map.entries(expected))
+  }
+};
+
+object Random {
+  var number = 4711;
+  public func next() : Nat {
+    number := (15485863 * number + 5) % 15485867;
```

Comment:
All other Random objects use 
```
number := (123138118391 * number + 133489131) % 9999;
```
formula.

Can you explain why your numbers are different? Is it better? Is it even matter?

Reply:
Yeah, I have found that their formula starts repeating the same sequence at some moment and I had a lot of similar sampled trees. Unfortunately, I didn't have time to investigate it.

Reply:
I have stolen this from `fuzz` library, enough I have used smaller primes.

Reply:
Ok, I'm fine with that

Reply:
To be honest this number looks very strange.
123138118391 and 9999 have common divisor 101
133489131 and 9999 have common divisor 3

I am not an expert in number theory, but It looks strange to me.

Reply:
I've check, my numbers works well enough